File: rest-service/api/src/main/java/org/veupathdb/service/multiblast/util/QueueDep.java
package org.veupathdb.service.multiblast.util;

import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;

import mb.lib.config.Config;
import org.apache.logging.log4j.Logger;
import org.veupathdb.lib.container.jaxrs.health.ServiceDependency;
import org.veupathdb.lib.container.jaxrs.providers.LogProvider;

public class QueueDep extends ServiceDependency
{
  private static final Logger log = LogProvider.logger(QueueDep.class);

  public static final String Name = "Job Queue";

  public QueueDep(Config conf) {
    super(Name, conf.getQueueHost(), 80);
  }

  @Override
  protected TestResult serviceTest() {
    try {
      var res = HttpClient.newBuilder()
        .connectTimeout(Duration.ofSeconds(2))
        .build()
        .send(
          HttpRequest.newBuilder().uri(URI.create(Address.http(getUrl()))).GET().build(),
          HttpResponse.BodyHandlers.discarding()
        );

      return new TestResult(this, true, Status.ONLINE);
    } catch (InterruptedException | IOException e) {
      log.warn("Dependency test failed", e);
      return new TestResult(this, true, Status.UNKNOWN);
    }
  }
}

File: rest-service/src/main/kotlin/mb/api/service/util/QueueDep.kt
package mb.api.service.util

import mb.lib.config.Config
import org.veupathdb.lib.container.jaxrs.health.Dependency
import org.veupathdb.lib.container.jaxrs.health.ServiceDependency
import org.veupathdb.lib.container.jaxrs.providers.LogProvider
import java.net.URI
import java.net.http.HttpClient
import java.net.http.HttpRequest
import java.net.http.HttpResponse
import java.time.Duration


object QueueDep: ServiceDependency("Job Queue", Config.queueHost, 80)
{
  val log = LogProvider.logger(QueueDep::class.java)

  override fun serviceTest() = try {
    HttpClient.newBuilder()
      .connectTimeout(Duration.ofSeconds(2))
      .build()
      .send(
        HttpRequest.newBuilder().uri(URI.create(Address.http(url))).GET().build(),
        HttpResponse.BodyHandlers.discarding()
      )

    Dependency.TestResult(this, true, Dependency.Status.ONLINE)
  } catch (e: Exception) {
    log.warn("Dependency test failed", e)
    Dependency.TestResult(this, true, Dependency.Status.UNKNOWN)
  }
}
